src/tree/v2/application/ApplicationResourceItem.ts

```diff
@@ -7,28 +7,34 @@ import { TreeItem } from 'vscode';
 import { BranchDataProvider, ResourceBase, ResourceModelBase } from '../../../api/v2/v2AzureResourcesApi';
 import { BranchDataItemCache } from '../BranchDataItemCache';
 import { BranchDataItemOptions, BranchDataProviderItem } from '../BranchDataProviderItem';
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
 
 export class ApplicationResourceItem<T extends ResourceBase> extends BranchDataProviderItem {
     constructor(
         public readonly resource: T,
         branchItem: ResourceModelBase,
         branchDataProvider: BranchDataProvider<ResourceBase, ResourceModelBase>,
         itemCache: BranchDataItemCache,
+        private readonly parent?: ResourceGroupsItem,
```

Comment:
Should it even be optional?

Reply:
Is the root considered the parent for top level tree nodes, like subscriptions? 

If it isn't, I could see parent being optional for whenever we do something like pinning/favorites.